src/main/java/nextstep/courses/domain/session/Session.java

```diff
@@ -13,47 +17,79 @@
+        CoverImage coverImage) {
+        this.id = id;
+        this.status = status;
+        this.metadata = new SessionMetadata(period, coverImage);
+        this.enrollment = new Enrollment(enrollmentPolicy);
     }
 
-    public Session(Long id, Period period, SessionStatus status) {
-        this(id, period, status, EnrollmentPolicy.free());
+    public static Session createFreeSession(Long id, Period period, CoverImage coverImage) {
+        return new Session(id, SessionStatus.PREPARING, new FreeEnrollmentPolicy(), period, coverImage);
     }
 
-    public Session(Long id, Period period, SessionStatus status, EnrollmentPolicy enrollmentPolicy) {
-        this(id, period, status, enrollmentPolicy, null);
+    public static Session createPaidSession(Long id, Period period, CoverImage coverImage, Amount price,
+        long capacity) {
+        return new Session(id, SessionStatus.PREPARING, new PaidEnrollmentPolicy(price, capacity), period, coverImage);
     }
 
-    public Session(Long id, Period period, SessionStatus status, EnrollmentPolicy enrollmentPolicy,
-        CoverImage coverImage) {
-        this.id = id;
-        this.metadata = new SessionMetadata(period, coverImage);
-        this.enrollment = new Enrollment(status, enrollmentPolicy);
+    public static Session restoreSession(Long id, SessionStatus status, Period period, CoverImage coverImage,
+        Amount price, long maxCapacity, int enrolledCount) {
+        EnrollmentPolicy policy;
+        if (price.equals(BigInteger.ZERO)) {
+            policy = new FreeEnrollmentPolicy();
+        } else {
+            policy = new PaidEnrollmentPolicy(price, maxCapacity);
+        }
+        Session session = new Session(id, status, policy, period, coverImage);
+        session.enrollment = new Enrollment(policy, enrolledCount);
+        return session;
     }
 
-    /* ------------ 정책 검증 ------------ */
-    // 모집중 상태인지
-    private boolean isOpen() {
-        return enrollment.isOpen();
+    /* 기능 */
+    public void open() {
+        validateNotEnded();
+        this.status = SessionStatus.OPEN;
     }
 
+    public void close() {
+        validateNotEnded();
+        this.status = SessionStatus.CLOSED;
+    }
+
+    private void validateNotEnded() {
+        LocalDate today = LocalDate.now();
+        if (today.isAfter(endAt())) {
+            throw new IllegalStateException("종료일 ( " + endAt() + " )이 지난 강의는 더이상 변경할 수 없습니다.");
+        }
+    }
+
+    /* ------------ 정책 검증 ------------ */
     // 수강신청이 가능한 "상태"인지 체크 (모집중 + 좌석 여유)
     public boolean canEnroll() {
-        return enrollment.canEnroll();
+        validateNotEnded();
+        if (!status.isOpen()) {
+            return false;
+        }
+        return enrollment.hasCapacity();
     }
 
     // 실제 수강신청
     public void enroll(Payment payment) {
+        if (!status.isOpen()) {
+            throw new CannotEnrollException("강의가 모집중이 아닙니다");
+        }
         enrollment.enroll(payment);
     }
```

Comment:
```suggestion
    public Enrollment enrollment() {
        // TODO Session 상태 값을 활용해 Enrollment 생성
    }
```

현재 구조는 Session이 Enrollment를 인스턴스 변수로 가지는 구조로 구현했다.
Session의 상태 값을 기반으로 Enrollment를 생성하는 메서드를 제공하는 것은 어떨까?